zerver/lib/openapi.py

```diff
@@ -25,28 +20,48 @@
     }
 }
 
+class OpenAPISpec():
+    def __init__(self, path: str) -> None:
+        self.path = path
+
+    def reload(self) -> None:
+        with open(self.path) as f:
+            yaml_parser = YamoleParser(f)
+        self.data = yaml_parser.data
+        self.last_update = os.path.getmtime(self.path)
```

Comment:
Move this line to the beginning of the function to avoid being stuck with an old version after a TOCTTOU race.